### PR TITLE
TASK: Add migration for ComponentContext deprecation

### PR DESCRIPTION
**Pull request by albe**
This adds some basic code migrations for the move away from HTTP Components.

Related: #2019, #2258

### CODE REVIEW

**Comment by albe** [comment]
Note: Not yet tested, see https://github.com/neos/neos-development-collection/pull/3175/files for referene on what should change. Will unfortunately not catch cases like https://github.com/neos/flow-development-collection/pull/2221/files#diff-35e8978565d45b4e9ee19751224b338fb75ad5bac258fd31ef9bdda976a11375 and might be too eager in case someone implemented a `getComponentContext()` method on his own class.

**Review by bwaidelich — CHANGES_REQUESTED** [review]
I just tested this with `Sitegeist.Monocle` and this migration removed all PHP code.. A bit harsh I would say :)

**Comment by bwaidelich** [comment]
I'll take care of a follow-up

**Review by bwaidelich — APPROVED** [review]
This seems to work now

File: Neos.Flow/Migrations/Code/Version20201207104500.php [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+    public function getIdentifier(): string
+    {
+        return 'Neos.Flow-20201207104500';
+    }
+
+    public function up(): void
+    {
+        $this->searchAndReplace('use Neos\Flow\Http\Component\SetHeaderComponent;', '', ['php']);
+        $this->searchAndReplaceRegex('/->setComponentParameter\((\\\\Neos\\\\Flow\\\\Http\\\\Component\\\\)?SetHeaderComponent::class, /', '->setHttpHeader(', ['php']);
```
Comment (bwaidelich) [inline Neos.Flow/Migrations/Code/Version20201207104500.php:34]:
```suggestion
        $this->searchAndReplaceRegex('/->setComponentParameter\((\\\\Neos\\\\Flow\\\\Http\\\\Component\\\\)?SetHeaderComponent::class, /',\s*'->setHttpHeader(', ['php']);
```

File: Neos.Flow/Migrations/Code/Version20201207104500.php [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+        $this->searchAndReplace('use Neos\Flow\Http\Component\ReplaceHttpResponseComponent;', '', ['php']);
+        $this->searchAndReplaceRegex('/->setComponentParameter\((\\\\Neos\\\\Flow\\\\Http\\\\Component\\\\)?ReplaceHttpResponseComponent::class, /', '->replaceHttpResponse(', ['php']);
```
Comment (bwaidelich) [inline Neos.Flow/Migrations/Code/Version20201207104500.php:36]:
```suggestion
        $this->searchAndReplaceRegex('/->setComponentParameter\((\\\\Neos\\\\Flow\\\\Http\\\\Component\\\\)?ReplaceHttpResponseComponent::class,\s*/', '->replaceHttpResponse(', ['php']);
```

**Pull request merged**